### PR TITLE
Serialization for `NipopowProof`

### LINKED ISSUE #490: [100 SigmaUSD] Add NipopowProof type
Scala reference implementation -
https://github.com/ergoplatform/ergo/blob/95139ec990fc91f8f4c70a61468edf8834143329/src/main/scala/org/ergoplatform/modifiers/history/popow/NipopowProof.scala#L25-L30
(ignore NipopowAlgos and skip NipopowProof methods)

In ergo-nipopow crate implement binary (implement ScorexSerializable trait) and JSON
serialization.

### PR DESCRIPTION
**Pull request by kettlebell**
Closes #490

### CODE REVIEW

File: ergo-rest/src/api/node.rs [hunk 1]
```diff
@@ -92,6 +92,6 @@ mod tests {
             timeout: Some(Duration::from_secs(5)),
         };
         let res = runtime_inner.block_on(async { get_info(node_conf).await.unwrap() });
-        assert_eq!(res.name, "ergo-mainnet-4.0.16.1");
```
Comment (kettlebell) [inline ergo-rest/src/api/node.rs:95]: The node now replies with "ew" as its name. We can't anticipate any future changes here, so check that it's not empty.

Comment (greenhat) [reply ergo-rest/src/api/node.rs:95]: Thanks! My bad, for making the test so fragile. I was very happy it worked so I forgot to make a more sensible check. :)

**Review by greenhat — APPROVED** [review]
Looking great! Thank you!

File: ergo-rest/src/api/node.rs (hunk 1 above)
Comment (kettlebell) [reply ergo-rest/src/api/node.rs:95]: I know the feeling!

**Pull request merged**